packages/headless/src/features/search-parameters/search-parameter-serializer.ts

```diff
@@ -158,13 +216,60 @@ function buildNumericRanges(ranges: string[]) {
     .map(([start, end]) => buildNumericRange({start, end, state: 'selected'}));
 }
 
+const relativeDateFormatRegexp = new RegExp(
+  // Matches `past1month`, `future5year`, etc.
+  `^(${validRelativeDatePeriods.join('|')})(\\d+)(${validRelativeDateUnits.join(
+    '|'
+  )})$`,
+  'i'
+);
+
+function isRelativeDateFormat(date: string) {
+  if (date.toLowerCase() === 'now') {
+    return true;
+  }
+
+  return relativeDateFormatRegexp.test(date);
+}
+
+function parseRelativeDateFormat(date: string): RelativeDateMap {
+  let relativeDate: RelativeDate;
+  if (date.toLowerCase() === 'now') {
+    relativeDate = {
+      period: 'now',
+    };
+  } else {
+    const matches = date.match(relativeDateFormatRegexp)!;
+    relativeDate = {
+      period: matches[1] as RelativeDatePeriod,
+      amount: parseInt(matches[2]),
+      unit: matches[3] as RelativeDateUnit,
+    };
+  }
+
+  return {...relativeDate, value: formatDate(relativeDate)};
```

Comment:
I'm having an issue here with the serialization. Since every time that we deserialize we have new absolute values, and the date-facet-set-slice does not contain the relative date, we cannot compare the values correctly. A workaround would be to provide the deserializer with the maps so he returns the previous absolute value instead of a new one.

Reply:
Or I would need to add the relative date maps to the search params 🤔